Adaptors/MongoDB/src/ServiceCollectionExt.cs

```diff
@@ -190,13 +169,37 @@ public static IServiceCollection AddMongoClient(this IServiceCollection services
     }
 
     var settings = MongoClientSettings.FromUrl(new MongoUrl(connectionString));
+
+    // Configure the connection settings
     settings.AllowInsecureTls       = mongoOptions.AllowInsecureTls;
     settings.UseTls                 = mongoOptions.Tls;
     settings.DirectConnection       = mongoOptions.DirectConnection;
     settings.Scheme                 = ConnectionStringScheme.MongoDB;
     settings.MaxConnectionPoolSize  = mongoOptions.MaxConnectionPoolSize;
     settings.ServerSelectionTimeout = mongoOptions.ServerSelectionTimeout;
     settings.ReplicaSetName         = mongoOptions.ReplicaSet;
+
+    if (!string.IsNullOrEmpty(mongoOptions.CAFile))
+    {
+      if (!File.Exists(mongoOptions.CAFile))
+      {
+        logger.LogError("CA certificate Mongo file not found at {path}",
+                        mongoOptions.CAFile);
+        throw new FileNotFoundException("CA certificate Mongo file not found",
+                                        mongoOptions.CAFile);
+      }
+
+      var (validationCallback, authority) = CertificateValidatorFactory.CreateCallback(mongoOptions.CAFile,
+                                                                                       logger);
+
+      settings.SslSettings = new SslSettings
+                             {
+                               ClientCertificates                  = new X509Certificate2Collection(authority),
```

Comment:
This is used for mTLS, but we do not support mTLS for now. It should most likely be unset.